src/controllers/krate/publish.rs

```diff
@@ -631,6 +602,37 @@ async fn count_versions_published_today(
         .await
 }
 
+async fn read_json_metadata(
+    reader: &mut (impl AsyncReadExt + Unpin),
+    max_length: u32,
+) -> Result<PublishMetadata, BoxedAppError> {
```

Comment:
I the `use tokio::io::AsyncReadExt` with the following should be enough?
```suggestion
async fn read_json_metadata<R: AsyncRead + Unpin>(
    reader: &mut R,
    max_length: u32,
) -> Result<PublishMetadata, BoxedAppError> {
```